plasmapy/classes/sources/openpmd_hdf5.py

```diff
@@ -8,6 +7,11 @@
 import os
 from distutils.version import StrictVersion
 
+# make h5py an optional dependency
+try:
+    import h5py
+except ImportError:
+    h5py = None
```

Comment:
I would raise another `ImportError` here, but with a message that is PlasmaPy specific (something like 'Using the openpmd_hdf5 module requires an installation of the hdf5 library').

Reply:
@dstansby Don't think that would solve the original problem. This will still give an error when importing `plasmapy` module.
```python
>>> import plasmapy
Traceback (most recent call last):
  File "/home/ritiek/PlasmaPy/plasmapy/classes/sources/openpmd_hdf5.py", line 12, in <module>
    import h5py
ModuleNotFoundError: No module named 'h5py'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ritiek/PlasmaPy/plasmapy/__init__.py", line 29, in <module>
    from . import classes
  File "/home/ritiek/PlasmaPy/plasmapy/classes/__init__.py", line 4, in <module>
    from . import sources
  File "/home/ritiek/PlasmaPy/plasmapy/classes/sources/__init__.py", line 3, in <module>
    from .openpmd_hdf5 import HDF5Reader
  File "/home/ritiek/PlasmaPy/plasmapy/classes/sources/openpmd_hdf5.py", line 15, in <module>
    raise ImportError("Using the openpmd_hdf5 module requires "
ImportError: Using the openpmd_hdf5 module requires an installation of the hdf5 library
```

Reply:
Ahhh, that's because importing `plasmapy` imports *everything* - is this deliberate/is there a good reason for doing this?

Reply:
It was sort of deliberate, but I wouldn't be sure about a good reason existing for this...